2_process/src/raster_in_polygon_weighted_mean.R

```diff
@@ -0,0 +1,47 @@
+raster_in_polygon_weighted_mean <- function(raster,
+                                            nhd_polygon_layer,
+                                            weighted_mean_col_name = 'dtb_weighted_mean',
+                                            comid_col = NULL){
+
+  #'@description function that takes a raster and vector polygon layer and summarizes the raster values within each polygon. Aggregation is a weighted mean. 
+  #'@param raster path to raster file (.tif or .adf), raster or SpatRaster object
+  #'@param nhd_polygon_layer polygon vector object. Polylines will not be accepted. Must buffer polylines before adding as input
+  #'@param weighted_mean_col_name col name for new summarized col name
+  #'@param comid_col comid col of the polygon. e.g 'comid' or 'featureid'. When NULL, No ID index is returned dataframe
```

Comment:
It's awesome that this function is generalizable to NHM, NHD, or other features! To make that more obvious, I'd vote for renaming `comid_col` to `feature_id` or `segment_id`. What do you think about that?